### PR TITLE
[Refactor] Refactor create table log process and Unified create table / view / mv

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 167 / 204 (81.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|0|3|00.00%|[567, 590, 609]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|2|00.00%|[744, 939]|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|0|1|00.00%|[95]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|4|7|57.14%|[252, 255, 258]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|30|46|65.22%|[774, 787, 1974, 1978, 1979, 1980, 1981, 1997, 1998, 2000, 2001, 2003, 3217, 3268, 4634, 4679]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|4|6|66.67%|[593, 1023]|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|52|61|85.25%|[218, 220, 521, 522, 523, 524, 535, 536, 552]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|11|12|91.67%|[145]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|40|40|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeMaterializedView.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableLikeAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/StorageVolumePEntryObject.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/starrocks/StarRocksDb.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/CreateTableInfo.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 164 / 212 (77.36%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|0|3|00.00%|[567, 590, 609]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|2|00.00%|[744, 939]|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|0|1|00.00%|[95]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|7|00.00%|[224, 225, 226, 227, 229, 230, 253]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|27|47|57.45%|[779, 792, 1972, 1978, 1979, 1980, 1981, 1983, 1986, 2004, 2005, 2007, 2008, 2010, 2015, 2033, 3166, 3217, 4577, 4622]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|4|7|57.14%|[252, 255, 258]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|4|6|66.67%|[593, 1023]|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|52|61|85.25%|[218, 220, 521, 522, 523, 524, 535, 536, 552]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|11|12|91.67%|[145]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|41|41|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeMaterializedView.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableLikeAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/StorageVolumePEntryObject.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/starrocks/StarRocksDb.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/CreateTableInfo.java|2|2|100.00%|[]|

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -1974,37 +1957,62 @@ void setLakeStorageInfo(OlapTable table, String storageVolumeId, Map<String, Str
...
         }
     }
 
     public void replayCreateTable(CreateTableInfo info) {
-        String dbName = info.getDbName();
         Table table = info.getTable();
-        Database db = this.fullNameToDb.get(dbName);
-        db.createTableWithLock(table, true);
+        Database db = this.fullNameToDb.get(info.getDbName());
+        db.writeUnlock();
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2005]: should be db.writeLock()

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 164 / 212 (77.36%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|0|3|00.00%|[567, 590, 609]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|2|00.00%|[744, 939]|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|0|1|00.00%|[95]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|7|00.00%|[224, 225, 226, 227, 229, 230, 253]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|27|47|57.45%|[779, 792, 1972, 1978, 1979, 1980, 1981, 1983, 1986, 2004, 2005, 2007, 2008, 2010, 2015, 2033, 3166, 3217, 4577, 4622]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|4|7|57.14%|[252, 255, 258]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|4|6|66.67%|[593, 1023]|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|52|61|85.25%|[218, 220, 521, 522, 523, 524, 535, 536, 552]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|11|12|91.67%|[145]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|41|41|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeMaterializedView.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableLikeAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/StorageVolumePEntryObject.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/starrocks/StarRocksDb.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/CreateTableInfo.java|2|2|100.00%|[]|

**Review by nshangyiming — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 2]
```diff
@@ -791,29 +787,16 @@ public boolean createTable(CreateTableStmt stmt) throws DdlException {
...
         if (tableFactory == null) {
-            ErrorReport.reportDdlException(ErrorCode.ERR_UNKNOWN_STORAGE_ENGINE, engineName);
+            ErrorReport.reportDdlException(ErrorCode.ERR_UNKNOWN_STORAGE_ENGINE, stmt.getEngineName());
         }
+
         Table table = tableFactory.createTable(this, db, stmt);
-        if (!(table instanceof OlapTable)) { // Special case: OlapTable has been added into the metastore before return.
-            registerTable(db, table, stmt);
-        }
+        String storageVolumeId = GlobalStateMgr.getCurrentState().getStorageVolumeMgr()
```
Comment (TszKitLo40) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:796]: `storageVolumeId` may be null in shared_nothing mode

File: fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java [hunk 3]
```diff
@@ -203,50 +200,37 @@ public Table createTable(LocalMetastore metastore, Database db, CreateTableStmt
...
+                String storageVolumeId = sv.getId();
                 metastore.setLakeStorageInfo(table, storageVolumeId, properties);
                 svm.bindTableToStorageVolume(sv.getId(), table.getId());
                 table.setStorageVolume(sv.getName());
             } else {
                 table = new OlapTable(tableId, tableName, baseSchema, keysType, partitionInfo, distributionInfo, indexes);
                 table.setStorageVolume(StorageVolumeMgr.LOCAL);
             }
-
-            if (table.isCloudNativeTable() && !runMode.isAllowCreateLakeTable())  {
```
Comment (TszKitLo40) [inline fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java:244]: I am not sure whether these two exception can be removed. @sduzh  PTAL

File: fe/fe-core/src/main/java/com/starrocks/server/StorageVolumeMgr.java [hunk 4]
```diff
@@ -228,13 +224,19 @@ public void unbindTableToStorageVolume(long tableId) {
...
+        }
+    }
+
     public String getStorageVolumeIdOfDb(long dbId) {
         try (LockCloseable lock = new LockCloseable(rwLock.readLock())) {
             return dbToStorageVolume.get(dbId);
         }
     }
 
-    public StorageVolume getDefaultStorageVolume() throws AnalysisException {
```
Comment (TszKitLo40) [inline fe/fe-core/src/main/java/com/starrocks/server/StorageVolumeMgr.java:237]: AnalysisException can also be removed in `createStorageVolume`, please also help me remove it.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 181 / 235 (77.02%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|0|3|00.00%|[567, 590, 609]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|2|00.00%|[744, 939]|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|0|1|00.00%|[95]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|7|00.00%|[224, 225, 226, 227, 229, 230, 253]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|27|49|55.10%|[343, 778, 791, 1971, 1977, 1978, 1979, 1980, 1982, 1985, 2004, 2005, 2007, 2008, 2010, 2015, 2033, 3166, 3217, 4577, 4622, 4628]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|4|7|57.14%|[252, 255, 258]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|4|6|66.67%|[593, 1023]|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|47|58|81.03%|[215, 217, 387, 388, 512, 513, 514, 515, 526, 527, 543]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|11|12|91.67%|[145]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|31|33|93.94%|[2302, 2303]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Table.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|41|41|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeMaterializedView.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableLikeAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/StorageVolumePEntryObject.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/starrocks/StarRocksDb.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/CreateTableInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|1|1|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
https://github.com/Mergifyio backport branch-3.1

**Pull request merged**